### PR TITLE
Add a demo and exercise around making/reviewing PRs

### LINKED ISSUE #103: Add walkthrough/exercise for opening, reviewing PRs
To _Collaborating with your Team_, add a long section (probably at least one hour, so maybe a whole new episode?) demonstrating how to open a pull request to fix an issue, and then how to review that, showcasing

- how to reference an issue, how to set up automatic closing of issue from PR
- marking as draft/ready for review
- exploring the diff
- commenting on a particular line
- suggesting changes
- approving/requesting changes/commenting

I have half an idea for a bot-based approach to do the setup for this activity in the long run, but for now I expect we can do our demo with a PR to address one of the many issues I have opened on this repository today!

### PR DESCRIPTION
**Pull request by tobyhodges**
Fixes #103 

The text for the PR activity is intentionally brief. It will be expanded after the first pilots, when we have created resources for a more consistent and scalable approach to this in future trainings (a template repository and branches, potentially automated issue opening, etc to provide a scaffold for Trainers to work on in the workshop).

### CODE REVIEW

File: episodes/26-collaborating.md [hunk 1]
```diff
@@ -57,6 +57,142 @@ you may also want to assign these issues to a member of your lesson development
...
+### Practice with Pull Requests (15 minutes)
+
+In breakout groups, assign yourselves the role of Submitter (S) and Reviewer (R),
+and repeat the process described above by completing the steps below.
+If anythere are any issues open on the repository that can be solved very quickly,
+feel free to address those with this exercise.
+Otherwise, the Submitter can open a Pull Request to make simple changes:
+typo fixes, or some completely arbitrary changes to a file.
+If the changes made for the exercise are not improvements to the lesson,
+the Reviewer should follow all the steps but close the PR at the end instead of merging it.
```
Comment (anenadic) [inline episodes/26-collaborating.md:175]: Not sure what we mean here - are we demonstrating a case when the PR is not accepted?

Comment (anenadic) [inline episodes/26-collaborating.md:175]: Not sure what we mean here - are we demonstrating the case when a PR is declined?

File: episodes/26-collaborating.md [hunk 2]
```diff
@@ -57,6 +57,142 @@ you may also want to assign these issues to a member of your lesson development
...
+_Note: This activity is designed to be carried out by two Trainers acting in two different roles:
+"(S)" denotes the Pull Request Submitter role, "(R)" the Pull Request Reviewer._
+
+1. \(S) Explore list of issues on repository, choose issue to fix, make note of issue number.
+1. \(S) Edit a file to fix the issue, commit changes to a new branch with an appropriate name.
+        Ensure that the changes include some typo etc that the Reviewer
+        will be able to comment on and/or suggest a fix/improvement to.
+1. \(S) Show how to open a pull request from that new branch to `main`.
+1. \(S) In Pull Request description, reference the number of the issue being fixed, 
+        using a keyword ("Fixes", "Resolves") to ensure automatic closure.
```
Comment (anenadic) [inline episodes/26-collaborating.md:138]:
```suggestion
        using a keyword "Fixes #X" or "Resolves #X" (where _X_ is the issue number) to ensure automatic closure. 
```

**Review by anenadic — COMMENTED** [review]
Looking really good @tobyhodges!

File: episodes/26-collaborating.md (hunk 1 above)
Comment (tobyhodges) [reply episodes/26-collaborating.md:175]: This is to cover the case where participants want to practice the PR workflow but do not have a specific improvement to make. In other words, if they are using a trivial change (adding "test test test test" to a line in a file, for example) to gain experience of the process and GitHub interface. We should allow people to do this - making real, good changes can take time! - but do not want to then include those non-helpful changes in the history of the `main` branch.

Or am I overthinking this?

**Comment by mkuzak** [comment]
Looks Good.
One question. Should protecting the `main` branch be done as a follow-along when it's demoed or should it be part of the exercise?

**Review by mkuzak — APPROVED** [review]

**Comment by tobyhodges** [comment]
> Should protecting the `main` branch be done as a follow-along when it's demoed or should it be part of the exercise?

What do you think is best? My plan was to do this as a follow-along.

**Pull request merged**